NeuralNetworkTrainer/src/neuralNetworkTrainer/EvolutionStrategy.java

```diff
@@ -1,43 +1,109 @@
+
+		for (ArrayList<ArrayList<Double>> individual : population) {
+			deserializedPopulation.add(Network.deserializeToNetwork(Driver.configuration, individual));
+		}
+
+		return deserializedPopulation;
+	}
+
+	// converts networks into matrixes for reproduction
+	public ArrayList<ArrayList<ArrayList<Double>>> serializePopulation(ArrayList<Network> population) {
+		ArrayList<ArrayList<ArrayList<Double>>> serializedPopulation = new ArrayList<ArrayList<ArrayList<Double>>>();
+
+		for (Network individual : population) {
+			serializedPopulation.add(Network.serializeNetwork(individual));
+		}
+
+		return serializedPopulation;
+	}
+
+	// creates a new generation using rank based selection of parents, crossover and
+	// mutation, then returns the new generation
+	// This is specific to having 2 offspring from 2 parents but is generalizable to
+	// any number of parents and offspring with a little refactoring
+	public ArrayList<ArrayList<ArrayList<Double>>> newGeneration(ArrayList<ArrayList<ArrayList<Double>>> population) {
 		//TODO
 		return null;
 	}
-	
-	public ArrayList<Network> selectOffspring(ArrayList<Network> population){
-		//TODO
+
+	public ArrayList<ArrayList<ArrayList<Double>>> selectParents(ArrayList<ArrayList<ArrayList<Double>>> population) {
+		// TODO
 		return null;
 	}
 
-	
-	public ArrayList<Network> crossoverOffspring(ArrayList<Network> offspring){
-		//TODO
+	// creates a List of boolean Lists that mirrors the dimensions and structure of
+	// an individual
+	// uses the list to randomly assign genes from parents to offspring
+	public ArrayList<ArrayList<ArrayList<Double>>> crossoverOffspring(ArrayList<ArrayList<Double>> parent1,
+			ArrayList<ArrayList<Double>> parent2) {
+		// TODO
 		return null;
 	}
-	
-	public ArrayList<Network> mutateOffspring(ArrayList<Network> offspring){
-		//TODO
+
+	// for any number of offspring:
+	// look at every Double and with a Driver.mutationRate chance
+	// change that value by getting a random gaussian distributed number centered at
+	// 0 with a standard deviation of 1
+	public ArrayList<ArrayList<ArrayList<Double>>> mutateOffspring(ArrayList<ArrayList<ArrayList<Double>>> offspring) {
+
+		// TODO
 		return null;
 	}
 
-	public Double evaluateFitness(Network network){
-		//TODO
+	// calls the correct fitness evaluation measure
+	private ArrayList<Network> evalFitness(ArrayList<Network> population) {
+		ArrayList<Network> fitPop = null;
+		if (Driver.isClassificationNetwork) {
+			fitPop = evalClasificationFitness(population);
+		} else {
+			fitPop = evalFunApproxFitness(population);
+		}
+		return fitPop;
+	}
+
+	private ArrayList<Network> evalFunApproxFitness(ArrayList<Network> population) {
+		// TODO Auto-generated method stub
 		return null;
 	}
-	
-	public Boolean hasConverged (ArrayList<Network> currentPopulation, ArrayList<Network> prevPopulation ){
-		//TODO
+
+	private ArrayList<Network> evalClasificationFitness(ArrayList<Network> population) {
+		// TODO Auto-generated method stub
```

Comment:
I think the evaluation for classification and function approx will be the same, no need for separate methods

Reply:
done